CONTRIBUTING.md

```diff
@@ -1,112 +1,124 @@
 
 
-**TERMINAL 1**
+## TERMINAL 1
 
 Run an *HTTP server* that will serve the payload. Then run `mosquitto` service.
 
 ```console
-$ nohup python3 -m http.server 8000 > /dev/null 2>&1 &
-$ mosquitto
+$ nohup python3 -m http.server --directory ./testdata 8000 > /dev/null 2>&1 &
+$ systemctl start mosquitto
 ```
 
-**TERMINAL 2**
+## TERMINAL 2
 
-Run `yggdrasil` specifying `mosquitto` as the MQTT server.
+Configure yggdrasil to connect to the local `moquitto` broker and increase the
+log level.
 
 ```console
-$ sudo go run ./cmd/yggd --server tcp://localhost:1883 --log-level trace --socket-addr @yggd
-...
-subscribed to topic: yggdrasil/localhost-22278168-85a6-11ec-a65c-fa163e3b5a61/data/in
-...
+cat /etc/yggdrasil/config.toml
+
+protocol = "mqtt"
+server = ["tcp://localhost:1883"]
+log-level = "debug"
 ```
-Or, if it is using via `rhc`.
 
-Configure in rhc the `moquitto` connection and the log level.
+Start the service:
 
 ```console
-cat /etc/rhc/config.toml
-# rhc global configuration settings
-
-broker = ["tcp://localhost:1883"]
-cert-file = "/etc/pki/consumer/cert.pem"
-key-file = "/etc/pki/consumer/key.pem"
-log-level = "bug"
+sudo systemctl start yggdrasil.service
 ```
 
-**TERMINAL 3**
+## TERMINAL 3
 
-Associate the `@yggd` socket with the `rhc-worker-playbook` worker.
+Assuming you built and installed the `rhc-worker-package` as described in
+[#Compilation](#compilation), start
+`com.redhat.Yggdrasil1.Worker1.rhc_worker_playbook.service`:
 
 ```console
-sudo YGG_SOCKET_ADDR=unix:@yggd python3 /usr/libexec/rhc/rhc-worker-playbook.worker
+sudo systemctl start com.redhat.Yggdrasil1.Worker1.rhc_worker_playbook.service
 ```
-This step is not needed if using the worker through `rhc`.
 
-**TERMINAL 4**
+## TERMINAL 4
 
-Execute the publish message in the "control/in" topic. Using `yggctl` to generate the data message with the Ansible playbook `test-module.yml` served by the HTTP server and `rhc-worker-playbook` as the directive.
+Generate a data message and publish it. `yggctl` (installed as part of the
+`yggdrasil` package) can easily generate data messages in the JSON format
+expected by `yggd`.
 
 ```console
- go run ./cmd/yggctl generate data-message --directive "rhc-worker-playbook" \"http://localhost:8000/test-module.yml\" | pub -broker tcp://localhost:1883 -topic yggdrasil/$CLIENT_ID/data/in
- ```
+echo '"http://localhost:8000/insights_remove.yml"' | \
+  yggctl generate data-message --directive rhc_worker_playbook -
+```
 
+This can then be piped to `pub` to publish the message:
 
- `$CLIENT_ID` can be found in the prompr of the TERMINAL 2, when `yggdrasil` is launched. The output tells the topic it is subscribed, in there the `$CLIENT_ID` can be found. Or in the logs of the `rhc`.
+```console 
+echo '"http://localhost:8000/insights_remove.yml"' | \
+  yggctl generate data-message --directive rhc_worker_playbook - | \
+  pub -broker tcp://localhost:1883 -topic yggdrasil/$(cat /var/lib/yggdrasil/client-id)/data/in
+```
```

Comment:
When I run this command, then I can see following in the log:

```
Aug 12 23:29:38 fedora yggd[14005]: [yggd] 2024/08/12 23:29:38 /builddir/build/BUILD/yggdrasil-0.4.2/_build/src/github.com/redhatinsights/yggdrasil/internal/http/client.go:48: sending HTTP request: GET Imh0dHA6Ly9sb2NhbGhvc3Q6ODAwMC9pbnNpZ2h0c19yZW1vdmUueW1sIgo=
Aug 12 23:29:38 fedora yggd[14005]: [yggd] 2024/08/12 23:29:38 /builddir/build/BUILD/yggdrasil-0.4.2/_build/src/github.com/redhatinsights/yggdrasil/internal/work/dispatcher.go:191: cannot dispatch data: cannot get detached message content: cannot do HTTP request: Get "Imh0dHA6Ly9sb2NhbGhvc3Q6ODAwMC9pbnNpZ2h0c19yZW1vdmUueW1sIgo=": unsupported protocol scheme ""
```

It looks like that there is something wrong with yggdrasil or worker. String "Imh0dHA6Ly9sb2NhbGhvc3Q6ODAwMC9pbnNpZ2h0c19yZW1vdmUueW1sIgo=" is base64 encoded "http://localhost:8000/insights_remove.yml", but it seems that something does not try to decode this string.

Reply:
```console
link@lima-fedora:.../Projects/rhc-worker-playbook$ yggctl -v
yggctl version 0.4.2
link@lima-fedora:.../Projects/rhc-worker-playbook$ echo '"http://localhost:8000/insights_remove.yml"' |   yggctl generate data-message --directive rhc_worker_playbook -
{"type":"data","message_id":"0fc2943d-9c1c-4d6e-b3a1-3442d17dd128","response_to":"","version":1,"sent":"2024-08-16T13:08:02.161028419-04:00","directive":"rhc_worker_playbook","metadata":{},"content":"http://localhost:8000/insights_remove.yml"}
```